packages/knip/src/IssueFixer.ts

```diff
@@ -76,22 +76,36 @@ export class IssueFixer {
   private async removeUnusedExportKeywords(issues: Issues) {
     const filePaths = new Set([...this.unusedTypeNodes.keys(), ...this.unusedExportNodes.keys()]);
     for (const filePath of filePaths) {
-      const exportPositions: Fixes = [
+      const fixes: Fixes = [
         ...(this.isFixUnusedTypes ? (this.unusedTypeNodes.get(filePath) ?? []) : []),
         ...(this.isFixUnusedExports ? (this.unusedExportNodes.get(filePath) ?? []) : []),
-      ].sort((a, b) => b[0] - a[0]);
+      ].sort((a, b) => b!.pos[0] - a!.pos[0]); // TODO how to handle types?
 
+      // TODO add type guard
+      const exportPositions = fixes;
       if (exportPositions.length > 0) {
-        const sourceFileText = exportPositions.reduce(
-          (text, [start, end]) => text.substring(0, start) + text.substring(end),
-          await readFile(filePath, 'utf-8')
-        );
-
-        const withoutEmptyReExports = sourceFileText
-          .replaceAll(/export \{[ ,]+\} from ('|")[^'"]+('|");?\r?\n?/g, '')
-          .replaceAll(/export \{[ ,]+\};?\r?\n?/g, '');
+        console.log(exportPositions);
+        const file = await readFile(filePath, 'utf-8');
+        const sourceFileText = exportPositions.reduce((text, fix) => {
+          if (isFixForExportWithElements(fix)) {
+            // TODO this could be called multiple times
+            const allElementFixesForExport = exportPositions
+              .filter(isFixForExportWithElements)
+              .filter(f => f.pos[0] === fix.pos[0] && f.pos[1] === fix.pos[1])
+              .map(f => f.element);
+
+            const allElements = fix.allElements.filter(element => !allElementFixesForExport.includes(element));
+            debugger;
+            if (!allElements.length) {
+              return text.substring(0, fix!.allExportPos[0]) + text.substring(fix!.allExportPos[1]);
+            }
+            return text.substring(0, fix.pos[0]) + `{${allElements.join(', ')}}` + text.substring(fix.pos[1]);
```

Comment:
This looks like we're re-assembling the exported items. How does that affect the original source code with regards to comments and whitespace?